src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Bucket<K, V>[] table;
+    private int size;
+    private int capacity;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Bucket[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
+
+    private class Bucket<K, V> {
+        private final K key;
+        private V value;
+        private Bucket<K, V> next;
+
+        public Bucket(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            increaseCapacity();
+        }
+        int index = getBucketIndex(key);
+        Bucket<K, V> bucket = table[index];
 
+        if (bucket == null) {
+            table[index] = new Bucket<>(key, value);
+        } else {
+            while (bucket.next != null) {
+                if (Objects.equals(bucket.key, key)) {
+                    bucket.value = value;
+                    return;
+                }
+                bucket = bucket.next;
+            }
+            if (Objects.equals(bucket.key, key)) {
+                bucket.value = value;
+                return;
+            }
+            bucket.next = new Bucket<>(key, value);
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getBucketIndex(key);
+        Bucket<K, V> bucket = table[index];
+
+        if (bucket == null) {
+            return null;
+        }
+        while (bucket != null) {
+            if (Objects.equals(bucket.key, key)) {
+                return bucket.value;
+            }
+            bucket = bucket.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void increaseCapacity() {
+        capacity *= 2;
```

Comment:
here we have a "magic number" which shoud be constant/ i'm not sure is it has large sence in this case , but i got the same comment from another our student , and i a bit agree with him , so , better to make a constant for this number)

Reply:
Ok, agree